app/models/post.rb

```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :author_id
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_save :update_post_counter
+
+  def five_most_recent_comments
+    comments.order(created_at: :desc).first(5)
+  end
+
+  def update_post_counter
+    author.increment!(:post_count)
+  end
```

Comment:
**OPTIONAL**

- [ ] Since you are using `after_save: update_post_counter` it is usually good practice to make the method itself private.